apps/artifactory/DEVHUB-README.md

```diff
@@ -6,17 +6,65 @@ Artifactory is an Artifact Repository system. It serves two primary purposes:
+Every project set is created with one service account to start, with the account name `default` - this means that your secret name will be `artifacts-default-[plate]` where the plate is random set of 6 alphanumeric characters.
+You can also find fuller details about your service account (including the current plate) by running the command `oc describe ArtifactoryServiceAccount default` or `oc describe artsvcacct default`
 
 For an easy way to get the secret information out via the CLI, try this command:
 
-`oc get secret/artifactory-serviceaccount-default -o json | jq '.data.password' | tr -d "\"" | base64 -d`
+`oc get secret/artifact-default-[plate] -o json | jq '.data.password' | tr -d "\"" | base64 -d`
+
+### Creating New Service Accounts
+
+Those with edit or admin access to their project set can create new Artifactory Service Account objects, and Docent will help set up all the various parts necessary to make it work!
+
+You can use the following command to do so:
+
+`oc process -f https://raw.githubusercontent.com/BCDevOps/developer-experience/master/apps/artifactory/artifactory-operator/config/samples/tmpl-artifactory-sa.yaml -p NAME="accountname" -p DESCRIPTOR="Description of Service Account" -p REGISTRIES='["key1","key2"]' | oc create -f -`
+
+Make sure you change the name and description parameters to suit your needs. 
+The registries parameter is optional - including it causes Docent to create a pull secret in addition to the normal secret for every docker registry key you include in the list.
+You will need to ensure that you are including only the registry keys, not the full registry url in this list; for example, `REGISTRIES='["docker-remote","redhat-docker-remote"]`.
+
+This will create an object in your namespace with whatever name you have given it. 
+You can find details about your object like this: `oc describe ArtifactoryServiceAccount [accountname]` or `oc describe artsvcacct [accountname]`
+and your relevant secret will be called `artifacts-[accountname]-[plate]` (as well as `artifacts-pull-[accountname]-[plate]` if you have chosen to include a registry list).
+
+### Deleting Service Accounts
+
+If you want to delete your service account for some reason, you can do so by deleting the ArtifactoryServiceAccount object through the Openshift CLI, like this:
+`oc delete ArtifactoryServiceAccount [accountname]` or `oc delete artsvcacct [accountname]`. 
+
+### I deleted my Artifactory Service Account secret(s)! What do I do now?!
+
+If you have deleted the secret for the default service account that is automatically created alongside your project set, 
+you just delete the ArtifactoryServiceAccount object called `default`, which you will find in your `tools` namespace, like so:
+
+`oc delete ArtifactoryServiceAccount default`
+
+The project provisioning bot will detect that this object has been deleted, and will recreate it automatically. No further action is required from you.
```

Comment:
Is there testing done with the race condition of deleting one CR before the manifest is sync'd back?  (re-creating the same object name via the operator may encounter race conditions?)

Reply:
A race condition is possible, I would think, but I don't think it's very likely, since the finalizer would be triggered first and the operator wouldn't have an opportunity to come back and notice the recreated CR until after the finalizer tasks are finished. Even if the CR did get recreated before the finalizer was finished, the operator should still act on both actions, in order.

I did do this several times for people who needed their secret recreated and never encountered this problem, but if you have any recommendations for better methods of testing other than "I tried a few times and it seemed okay" (lol) I would be definitely up for giving them a go!

Reply:
Also I just talked it out with Justin a bit, and deleting a CR marks it for deletion first, then runs the finalizer, and only then does the object actually get deleted. This means that the project provisioner won't attempt to recreate the object until after that point, so it can't create a CR to create a new version of the object if the finalizer is still running.